### PR TITLE
Fix mis-detecting network configuration in initramfs cmdline

### PR DESCRIPTION
**Pull request by TheRealFalcon**
**Integration test still needs to be written**

## Proposed Commit Message

```
Fix mis-detecting network configuration in initramfs cmdline

klibc initramfs in debian allows the 'iscsi_target_ip=' cmdline
parameter to specify an iscsi device attachment. This can
cause cloud-init to mis-detect the cmdline paramter as a
networking config.

LP: #1919188
```

## Additional Context
https://bugs.launchpad.net/cloud-init/+bug/1919188

## Test Steps
**Integration test still needs to be written**

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/hacking.html)
 - [x] I have updated or added any unit tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

File: cloudinit/net/cmdline.py
Comment (OddBloke) [inline cloudinit/net/cmdline.py:76]: I think `[6]*` is slightly incorrect, it would match, e.g., `ip666` (the addressing protocol of the beast); `6?` will match 0 or 1 "6"s. (Either way, the square brackets are superfluous: a character and a bracket expression containing only that character are equivalent.) This is borderline nit territory, but as regexes are hard enough to read at the best of times, I think it's worth being exact here.

I think we're also missing tests for the `^` choice of the subexpression here: nothing fails if I edit the regex to only have `\s` there instead of the subexpression. Can you add a test for that?

Comment (mwhudson) [reply cloudinit/net/cmdline.py:76]: I'm just flying past here, but you should really "parse" the kernel cmdline with shlex.split, because that's essentially what the initramfs is doing. This regex would match something like the admittedly extremely contrived `opt="bunch of stuff with spaces and ip=foo"`. Also the fewer regexes in the codebase the better!

**Comment by TheRealFalcon** [comment]
@OddBloke Updated based on review comments.

I attempted writing an integration test for this behavior, but I was unable to create one. It requires
1. Passing kernel command line arguments
2. The datasource indicating there is no network config

#1 is doable but hacky. I can't figure out a way to do #2. Can you think of a realistic way to do #2? If not, are you ok having just unit test coverage for this one?

**Comment by TheRealFalcon** [comment]
Updated (as a separate commit) to do this using shlex instead.

**Comment by mwhudson** [comment]
Thanks, that makes me much less queasy :-)

I think it would make sense to parse the kernel command line into a dict more generally but that's a much bigger change of course.

**Review by OddBloke — APPROVED** [review]
This LGTM, thanks! I wonder if we should add a test for quoted content (i.e. cmdline of `foo="ip=shouldntbedetected"`); I'll let you decide and land this with or without it.

**Comment by TheRealFalcon** [comment]
Separate from this one? https://github.com/canonical/cloud-init/pull/844/files#diff-402c7850e179c77acffc2f2a35c464a2ba0bdb6517060dbf8a9f6a6a571432b3R4435

**Comment by OddBloke** [comment]
> Separate from this one? https://github.com/canonical/cloud-init/pull/844/files#diff-402c7850e179c77acffc2f2a35c464a2ba0bdb6517060dbf8a9f6a6a571432b3R4435

That looks good, no idea how I missed it first time through!

**Pull request merged**
